perception/detected_object_validation/src/object_lanelet_filter.cpp

```diff
@@ -0,0 +1,184 @@
+      /*target*/ target_frame_id, /*src*/ source_frame_id, time,
+      rclcpp::Duration::from_seconds(0.5));
+    return self_transform_stamped.transform;
+  } catch (tf2::TransformException & ex) {
+    RCLCPP_WARN_STREAM(rclcpp::get_logger("object_lanelet_filter"), ex.what());
+    return boost::none;
+  }
+}
+namespace object_lanelet_filter
+{
+ObjectLaneletFilterNode::ObjectLaneletFilterNode(const rclcpp::NodeOptions & node_options)
+: Node("object_lanelet_filter_node", node_options),
+  tf_buffer_(this->get_clock()),
+  tf_listener_(tf_buffer_)
+{
+  using std::placeholders::_1;
+
+  // Set parameters
+  unknown_only_ = declare_parameter<bool>("unknown_only", false);
+
+  // Set publisher/subscriber
+  map_sub_ = this->create_subscription<autoware_auto_mapping_msgs::msg::HADMapBin>(
+    "input/vector_map", rclcpp::QoS{1}.transient_local(),
+    std::bind(&ObjectLaneletFilterNode::mapCallback, this, _1));
+  object_sub_ = this->create_subscription<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "input/object", rclcpp::QoS{1}, std::bind(&ObjectLaneletFilterNode::objectCallback, this, _1));
+  object_pub_ = this->create_publisher<autoware_auto_perception_msgs::msg::DetectedObjects>(
+    "output/object", rclcpp::QoS{1});
+}
+
+void ObjectLaneletFilterNode::mapCallback(
+  const autoware_auto_mapping_msgs::msg::HADMapBin::ConstSharedPtr map_msg)
+{
+  lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(*map_msg, lanelet_map_ptr_);
+  const lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(lanelet_map_ptr_);
+  road_lanelets_ = lanelet::utils::query::roadLanelets(all_lanelets);
+}
+
+void ObjectLaneletFilterNode::objectCallback(
+  const autoware_auto_perception_msgs::msg::DetectedObjects::ConstSharedPtr input_msg)
+{
+  using Label = autoware_auto_perception_msgs::msg::ObjectClassification;
+
+  // Guard
+  if (object_pub_->get_subscription_count() < 1) return;
+
+  autoware_auto_perception_msgs::msg::DetectedObjects output_object_msg;
+  output_object_msg.header = input_msg->header;
+
+  if (!lanelet_map_ptr_) {
+    RCLCPP_ERROR(get_logger(), "No vector map received.");
+    return;
+  }
+  autoware_auto_perception_msgs::msg::DetectedObjects transformed_objects;
+  if (!transformDetectedObjects(*input_msg, "map", tf_buffer_, transformed_objects)) {
+    RCLCPP_ERROR(get_logger(), "Failed transform to map.");
+    return;
+  }
+
+  // calculate convex hull
+  const auto convex_hull = getConvexHull(transformed_objects);
+  // get intersected lanelets
+  lanelet::ConstLanelets intersected_lanelets = getIntersectedLanelets(convex_hull, road_lanelets_);
+
+  int index = 0;
+  for (const auto & object : transformed_objects.objects) {
+    const auto position = object.kinematics.pose_with_covariance.pose.position;
+    const auto & label = object.classification.front().label;
+    if (unknown_only_ && (label != Label::UNKNOWN)) {
```

Comment:
we want to remove vehicle outside of lanelet2.
so, it may be better to set for each object.
sample : 
```
ignore:
  CAR : true
  PEDESTRIAN : false
  UNKNOWN : true
```

Reply:
I see. Like this?
param
```
    filter_target_label:
      UNKNOWN : true
      CAR : false
      TRUCK : false
      BUS : false
      TRAILER : false
      MOTORCYCLE : false
      BICYCLE : false
      PEDESTRIAN : false
```
code
```
    if ((label != Label::UNKNOWN && filter_target_.UNKNOWN) || 
      (label != Label::CAR && filter_target_.CAR)||
      (label != Label::TRUCK && filter_target_.TRUCK)||
      (label != Label::BUS && filter_target_.BUS)||
      (label != Label::TRAILER && filter_target_.TRAILER)||
      (label != Label::MOTORCYCLE && filter_target_.MOTORCYCLE)||
      (label != Label::BICYCLE && filter_target_.BICYCLE)||
      (label != Label::PEDESTRIAN && filter_target_.PEDESTRIAN)) {
```

Reply:
right :+1:

Reply:
Add option for each class
https://github.com/autowarefoundation/autoware.universe/pull/1221/commits/219a862b3497bc899ce430ee0b2b2b37378f4928